pkg/simulation/target.go

```diff
@@ -92,3 +94,29 @@ func (sim *Simulation) Neutrals() []key.TargetID {
 	copy(out, sim.neutrals)
 	return out
 }
+
+func (sim *Simulation) Retarget(data info.Retarget) []key.TargetID {
+	// merged removal of filtered targets and targets in limbo into 1 loop.
+	i := 0
+	for _, target := range data.Targets {
+		if (data.IncludeLimbo || sim.Attr.HPRatio(target) > 0) && data.Filter(target) {
+			data.Targets[i] = target
+			i++
+		}
+	}
```

Comment:
In my original comment, forgot to mention that you need to truncate this list after the loop
```go
data.Targets = data.Targets[:i]
```